backend/internal/model/game_test.go

```diff
@@ -138,6 +138,97 @@ func TestGameSubGoal(t *testing.T) {
 	}
 }
 
+func Test_game_UpdateManualGoals(t *testing.T) {
+	game := &game{score: GameScore{0, 0}, manualGoals: ManualGoals{0, 0, 0, 0}}
+
+	type args struct {
+		name                string
+		teamID              int
+		action              string
+		expectedManualGoals ManualGoals
+		expectedError       string
+	}
+	tests := []args{
+		// No error cases
+		{
+			name:                "should increment blue manual goals by 1",
+			teamID:              config.TeamBlue,
+			action:              config.ActionAdd,
+			expectedManualGoals: ManualGoals{1, 0, 0, 0},
+			expectedError:       "",
+		},
+		{
+			name:   "should decrement blue manual goals by 1",
+			teamID: config.TeamBlue, action: config.ActionSubtract,
+			expectedManualGoals: ManualGoals{0, 1, 0, 0},
+			expectedError:       "",
+		},
+		{
+			name:   "should increment white manual goals by 1",
+			teamID: config.TeamWhite, action: config.ActionAdd,
+			expectedManualGoals: ManualGoals{0, 0, 1, 0},
+			expectedError:       "",
+		},
+		{
+			name:   "should decrement white manual goals by 1",
+			teamID: config.TeamWhite, action: config.ActionSubtract,
+			expectedManualGoals: ManualGoals{0, 0, 0, 1},
+			expectedError:       "",
+		},
+
+		// Error cases
+		{
+			name:   "should return bad team ID error",
+			teamID: 0, action: config.ActionAdd,
+			expectedManualGoals: ManualGoals{0, 0, 0, 0},
+			expectedError:       "bad team ID",
+		},
+		{
+			name:   "should return bad team ID error",
+			teamID: 0, action: config.ActionSubtract,
+			expectedManualGoals: ManualGoals{0, 0, 0, 0},
+			expectedError:       "bad team ID",
+		},
+		{
+			name:   "should return bad action error",
+			teamID: config.TeamBlue, action: "addd",
+			expectedManualGoals: ManualGoals{0, 0, 0, 0},
+			expectedError:       "bad action type",
+		},
+		{
+			name:                "should return bad action error",
+			teamID:              config.TeamWhite,
+			action:              "addd",
+			expectedManualGoals: ManualGoals{0, 0, 0, 0},
+			expectedError:       "bad action type",
+		},
+		{
+			name:                "should return bad action error",
+			teamID:              -10,
+			action:              "addd",
+			expectedManualGoals: ManualGoals{0, 0, 0, 0},
+			expectedError:       "bad action type",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			game.manualGoals = ManualGoals{0, 0, 0, 0}
+			err := game.UpdateManualGoals(tt.teamID, tt.action)
+
+			if tt.expectedError == "" {
+				assert.Nil(t, err)
+			} else {
+				assert.EqualError(t, err, tt.expectedError)
+			}
+
+			assert.Equal(t, game.manualGoals.AddedBlue, tt.expectedManualGoals.AddedBlue, "Manual goals for blue team added incorrectly")
```

Comment:
Why are you comparing each subvalue separately? Is `assert.Equal(t, game.manualGoals, tt.expectedManualGoals, "Manual goals added incorrectly")` not working?

Reply:
Good point, we have changed it